src/bio/bio_monitor.c

```diff
@@ -490,5 +558,8 @@ bio_media_error(void *msg_arg)
 	/* TODO Implement checksum error counter */
 	dev_state->bds_checksum_errs = 0;
 
+	/* Notify admin through Control Plane of BIO error */
+	notify_bio_error(mem->mem_unmap, mem->mem_update, mem->mem_tgt_id);
```

Comment:
I'm afraid this won't work since this sync dRPC call could block the hardware poll ULT.

Actually I think explicit calling to dRPC code in BIO module is somewhat layer violation, you could make it as a callback and put it in bio_reaction_ops.

Reply:
Also, I think we don't need to notify control plane on every BIO or checksum error, right? I think we only need to notify control plane when the device state is changed, what do you think?

Reply:
I thought the first step of faulty device detection was notifying the user upon a BIO error, and then allowing the admin the ability to manually set the device state to FAULTY. so notifying only upon device state change would not make sense in the manual mode right? maybe the automated method of a device state being set to FAULTY and then notifying admin

Reply:
What about the option of adding another state - something like "BIO_BS_STATE_ERR_DETECTED", and then adding a callback for that which would be to notify the admin thru the control plane/dRPC?

Reply:
Right, notifying admin the IO errors could be useful in this case, I overlooked it.

I don't think we need to introduce a new state, just adding a callback for notifying control plane will be fine.